src/App.jsx

```diff
@@ -3,46 +3,62 @@ import React, { useState, useEffect } from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = useState(
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  );
-
-  useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-  }, [todoList]);
-
-  return [todoList, setTodoList];
-}
-
 function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('todoList')) ?? []);
+  const [isLoading, setIsLoading] = useState(true);
 
+  // Load initial data and save to localStorage
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+        const options = {
+          method: 'GET',
+          headers: {
+            Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`
+          }
+        };
+        const response = await fetch(url, options);
+        if (!response.ok) {
+          throw new Error('Error: ${response.status}');
```

Comment:
I would take a look at this `'Error: ${response.status}'` again and see what it returns as opposed to what you expect it to return.

Reply:
Do not use inverted commas ` throw new Error(`Error: ${response.status}`);`

Reply:
So you were using commas `'` instead of backticks ```. Just look out for that distinction when working with string interpolations. All you need to change are the commas to ``Error: ${response.status}``